azext_iot/tests/iothub/devices/test_hub_control_plane_export_import_int.py

```diff
@@ -0,0 +1,305 @@
+import random
+import pytest
+import json
+import os
+import time
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.tests.settings import DynamoSettings, ENV_SET_TEST_IOTHUB_REQUIRED, ENV_SET_TEST_IOTHUB_OPTIONAL
+from azure.cli.command_modules.iot.tests.latest._test_utils import _create_test_cert, _delete_test_cert
+from azext_iot.tests.iothub import IoTLiveScenarioTest
+from azext_iot.tests.test_constants import ResourceTypes
+from azext_iot.tests.generators import generate_generic_id
+from azext_iot.tests.helpers import add_test_tag
+from azext_iot.tests.iothub import DATAPLANE_AUTH_TYPES
+
+settings = DynamoSettings(req_env_set=ENV_SET_TEST_IOTHUB_REQUIRED, opt_env_set=ENV_SET_TEST_IOTHUB_OPTIONAL)
+CWD = os.path.dirname(os.path.abspath(__file__))
+
+CERT_FILE = "test_cert.cer"
+KEY_FILE = "test_key.cer"
+DATAPLANE_AUTH_TYPES.remove("cstring")
+
+
+class TestHubControlPlaneExportImport(IoTLiveScenarioTest):
+    def __init__(self, test_case):
+        super(TestHubControlPlaneExportImport, self).__init__(test_case)
+        self.dest_hub = settings.env.azext_iot_desthub or "test-hub-" + generate_generic_id()
+        self.dest_hub_rg = settings.env.azext_iot_destrg or settings.env.azext_iot_testrg
+        self.cli = EmbeddedCLI()
+
+        hub_id = self.cli.invoke(f"iot hub show -n {self.entity_name} -g {self.entity_rg}").as_json()["id"]
+        self.subscription_id = hub_id.split("/")[2]
+
+        # create destination hub
+
+        if not settings.env.azext_iot_desthub:
+            self.create_hub(self.dest_hub, self.dest_hub_rg)
+
+        add_test_tag(
+            cmd=self.cmd,
+            name=self.dest_hub,
+            rg=self.dest_hub_rg,
+            rtype=ResourceTypes.hub.value,
+            test_tag=test_case
+        )
+
+        self.dest_hub_cstring = self.cmd(
+            f"iot hub connection-string show -n {self.dest_hub} -g {self.dest_hub_rg}"
+        ).get_output_in_json()["connectionString"]
+
+        self.filename = generate_generic_id() + ".json"
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_class(self):
+
+        self.clean_up_hub(self.entity_name, self.entity_rg)
+
+        # add identities
+        self.uid = "/subscriptions/a386d5ea-ea90-441a-8263-d816368c84a1/resourcegroups/mirabai/providers/" + \
```

Comment:
We do not want to hard-code any resources/infrastructure in the tests themselves. Ideally they should be provisioned in JIT fashion as part of the test setup. Furthermore something like a subscription Id is an internal identifier and this is a public project.

A mode 2 for tests that we often support, is usage of environment variables that point to an existing resource i.e. if a target resource is provided via environment variable use it (and skip deleting it if user provided), otherwise provision & configure a new resource, then delete if it was provisioned for the test.